database/sqlite/sqlite_aclk_alert.c

```diff
@@ -0,0 +1,537 @@
+{
+    if (unlikely(!node_id))
+        return;
+
+    struct aclk_database_worker_config *wc  = NULL;
+    struct aclk_database_cmd cmd;
+    memset(&cmd, 0, sizeof(cmd));
+    cmd.opcode = ACLK_DATABASE_ALARM_HEALTH_LOG;
+
+    rrd_wrlock();
+    RRDHOST *host = find_host_by_node_id(node_id);
+    if (likely(host))
+        wc = (struct aclk_database_worker_config *)host->dbsync_worker;
+    rrd_unlock();
+    if (wc)
+        aclk_database_enq_cmd(wc, &cmd);
+    else {
+        if (aclk_worker_enq_cmd(node_id, &cmd))
+            error_report("ACLK synchronization thread is not active for node id %s", node_id);
+    }
+    return;
+}
+
+void aclk_push_alarm_health_log(struct aclk_database_worker_config *wc, struct aclk_database_cmd cmd)
+{
+    UNUSED(cmd);
+#ifndef ACLK_NG
+    UNUSED(wc);
+#else
+    int rc;
+
+    char *claim_id = is_agent_claimed();
+    if (unlikely(!claim_id))
+        return;
+
+    uint64_t first_sequence = 0;
+    uint64_t last_sequence = 0;
+    struct timeval first_timestamp;
+    struct timeval last_timestamp;
+
+    BUFFER *sql = buffer_create(1024);
+
+    sqlite3_stmt *res = NULL;
+
+    //TODO: make this better: include info from health log too
+    buffer_sprintf(sql, "select aa.sequence_id, aa.date_created, \
+                         (select laa.sequence_id from aclk_alert_%s laa \
+                         order by laa.sequence_id desc limit 1), \
+                         (select laa.date_created from aclk_alert_%s laa \
+                         order by laa.sequence_id desc limit 1) \
+                         from aclk_alert_%s aa order by aa.sequence_id asc limit 1;", wc->uuid_str, wc->uuid_str, wc->uuid_str);
+
+    rc = sqlite3_prepare_v2(db_meta, buffer_tostring(sql), -1, &res, 0);
+    if (rc != SQLITE_OK) {
+        error_report("Failed to prepare statement to get health log statistics from the database");
+        buffer_free(sql);
+        freez(claim_id);
+        return;
+    }
+
+    first_timestamp.tv_sec = 0;
+    first_timestamp.tv_usec = 0;
+    last_timestamp.tv_sec = 0;
+    last_timestamp.tv_usec = 0;
+
+    while (sqlite3_step(res) == SQLITE_ROW) {
+        first_sequence = sqlite3_column_bytes(res, 0) > 0 ? (uint64_t) sqlite3_column_int64(res, 0) : 0;
+        if (sqlite3_column_bytes(res, 1) > 0) {
+            first_timestamp.tv_sec = sqlite3_column_int64(res, 1);
+        }
+
+        last_sequence = sqlite3_column_bytes(res, 2) > 0 ? (uint64_t) sqlite3_column_int64(res, 2) : 0;
+        if (sqlite3_column_bytes(res, 3) > 0) {
+            last_timestamp.tv_sec = sqlite3_column_int64(res, 3);
+        }
+    }
+
+    struct alarm_log_entries log_entries;
+    log_entries.first_seq_id = first_sequence;
+    log_entries.first_when = first_timestamp;
+    log_entries.last_seq_id = last_sequence;
+    log_entries.last_when = last_timestamp;
+
+    struct alarm_log_health alarm_log;
+    alarm_log.claim_id = claim_id;
+    alarm_log.node_id = get_str_from_uuid(wc->host->node_id);
```

Comment:
you can either use `wc->node_id` here or you need to release the memory around L333

Reply:
Ok, changed in https://github.com/netdata/netdata/pull/11552/commits/2194985e45fab658c6446750847fe3ca5e93e823

Reply:
Ok, can use wc->node_id by itself maybe too and get rid of a strdupz/freez, we can do it both in charts too, after the merge maybe?